providers/async_registry.py

```
@@ -0,0 +1,310 @@
+
+        # Get async provider class
+        provider_class = instance._async_providers[provider_type]
+
+        # Initialize async provider based on type
+        try:
+            if provider_type == ProviderType.CUSTOM:
+                # Handle custom providers
+                custom_url = os.getenv("CUSTOM_API_URL", "")
+                if not custom_url:
+                    if api_key:  # Key is set but URL is missing
+                        logger.warning("CUSTOM_API_KEY set but CUSTOM_API_URL missing – skipping Custom async provider")
```

Pylint (3.3.7)

[convention] Line too long (120/100)

(C0301)